keps/sig-cli/4292-kubectl-debug-custom-profile/README.md

```diff
@@ -0,0 +1,1001 @@
+		}
+	} else {
+		debugContainerJS, err = json.Marshal(debugPod.Spec.Containers[index])
+		if err != nil {
+			return fmt.Errorf("unable to marshall ephemeral container:%v", err)
+		}
+	}
+
+	patchedContainer, err := strategicpatch.StrategicMergePatch(debugContainerJS, customJS, corev1.Container{})
+	if err != nil {
+		return fmt.Errorf("error creating three way patch to add debug container: %v", err)
+	}
+
+	if ephemeral {
+		err = json.Unmarshal(patchedContainer, &debugPod.Spec.EphemeralContainers[index])
+		if err != nil {
+			return fmt.Errorf("unable to unmarshall patched container to ephemeral container: %v", err)
+		}
+	} else {
+		err = json.Unmarshal(patchedContainer, &debugPod.Spec.Containers[index])
+		if err != nil {
+			return fmt.Errorf("unable to unmarshall patched container to ephemeral container: %v", err)
+		}
+	}
+	return nil
+}
+```
+
+After predefined profiles are applied, there will be a generated podSpec that is pending for creation.
+Before sending a request to API server to create this pod or ephemeral container, `applyCustomProfile` function will be called
+with the required parameters. This function modifies podSpec internally and this podSpec will be created/updated.
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/debug`: `13-10-2023` - `62.7%`
```

Comment:
Can you describe the planned additions to units?

Reply:
Not addressed

Reply:
Test cases are added.